libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -111,19 +111,19 @@ fn format_function_expression() {
 
     check(
         script,
-        expect![[r#"
+        expect![[r##"
             package main
             (a) => {
-                return a:A
-            }:(a:A) => A
+                return a:#A
+            }:(a:#A) => #A
```

Comment:
Reading the code `#` means the variable is free. Why is `A` considered free here? My expectation is that `A` would be bound to the polytype of the function. What and I misunderstanding?

Reply:
It should be, but the way the code works now we perform generalizations when it is bound to a variable, not when the function is created. I don't think this is entirely correct, but I didn't want to change the behavior here.